### ISSUE TITLE
$<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_LIBRARIES>>

### ISSUE THREAD

**Issue by lucaszanella** [AndrewGaspar/cmake-cargo#23]
There were no instructions os I simply cloned this repo along my CMakeLists.txt like this:

`CMakeLists.txt cmake-cargo Cargo.toml build`

This is the `CMakeLists.txt`:

```
cmake_minimum_required (VERSION 3.10)
project(CargoTest LANGUAGES CXX)

list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake-cargo/cmake")
set(CARGO_DEV_MODE ON CACHE INTERNAL "")

include(Cargo)

add_crate(Cargo.toml)
```

it should build my `Cargo.toml`, shouldn't it?

so inside build I did `cmake ../`

```
root@e85658e77489:/workspaces/smoltcp_openvpn_bridge/build# cmake ../
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Rust Target: x86_64-unknown-linux-gnu
-- Found Cargo: /root/.cargo/bin/cargo (found version "1.44.0") 
-- Defaulting Cargo to build debug
-- Running in cmake-cargo dev mode
-- Configuring done
CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_LIBRARIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_DIRECTORIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_LIBRARIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_DIRECTORIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_LIBRARIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


CMake Error at cmake-cargo/cmake/Cargo.cmake:36 (add_custom_target):
  Error evaluating generator expression:

    $<GENEX_EVAL:$<TARGET_PROPERTY:cargo-build_smoltcp_cpp_interface_rust,CARGO_LINK_DIRECTORIES>>

  Expression did not evaluate to a known generator expression
Call Stack (most recent call first):
  build/CMakeFiles/cmake-cargo/smoltcp_openvpn_bridge.dir/cargo-build.cmake:3 (_add_cargo_build)
  cmake-cargo/cmake/Cargo.cmake:115 (include)
  CMakeLists.txt:9 (add_crate)


-- Generating done
-- Build files have been written to: /workspaces/smoltcp_openvpn_bridge/build
```

If you wanna take a look at the project: https://github.com/lucaszanella/smoltcp_cpp_interface/tree/190fc6c3bfa426b00e28361599fdb669a527cd94

**Comment by AndrewGaspar** [comment]
Ah, it turns out cmake-cargo has a dependency on Cmake 3.12 or newer. You'll need to install a newer CMake version. If I had to take a guess, you're using a Ubuntu 18.04 image and apt-get to install cmake? That version is 3.10, which is fairly out of date now.

I'll update cmake-cargo to indicate it requires `cmake_minimum_required(VERSION 3.12)`.

**Comment by AndrewGaspar** [comment]
Btw, I'm working towards improving the documentation on this. Currently blocked on another open source project, but I'll focus more intently on this in the next week or so.

**Comment by AndrewGaspar** [comment]
Fixed in dc95ec7442bad8e3b8b5a160bcd51b3752ad6512
